File: software/examples/rust/example_threshold.rs
use std::{error::Error, io, thread};
use tinkerforge::{accelerometer_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Accelerometer Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let a = AccelerometerBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get threshold receivers with a debounce time of 10 seconds (10000ms).
    a.set_debounce_period(10000);

    let acceleration_reached_receiver = a.get_acceleration_reached_callback_receiver();

    // Spawn thread to handle received callback messages.
    // This thread ends when the `a` object
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for acceleration_reached in acceleration_reached_receiver {
            println!("Acceleration [X]: {} g", acceleration_reached.x as f32 / 1000.0);
            println!("Acceleration [Y]: {} g", acceleration_reached.y as f32 / 1000.0);
            println!("Acceleration [Z]: {} g", acceleration_reached.z as f32 / 1000.0);
            println!();
        }
    });

    // Configure threshold for acceleration "greater than 2 g, 2 g, 2 g".
    a.set_acceleration_callback_threshold('>', 2 * 1000, 0, 2 * 1000, 0, 2 * 1000, 0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/accelerometer_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Accelerometer Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	a, _ := accelerometer_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get threshold receivers with a debounce time of 10 seconds (10000ms).
	a.SetDebouncePeriod(10000)

	a.RegisterAccelerationReachedCallback(func(x int16, y int16, z int16) {
		fmt.Printf("Acceleration [X]: %f g\n", float64(x)/1000.0)
		fmt.Printf("Acceleration [Y]: %f g\n", float64(y)/1000.0)
		fmt.Printf("Acceleration [Z]: %f g\n", float64(z)/1000.0)
		fmt.Println()
	})

	// Configure threshold for acceleration "greater than 2 g, 2 g, 2 g".
	a.SetAccelerationCallbackThreshold('>', 2*1000, 0, 2*1000, 0, 2*1000, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
